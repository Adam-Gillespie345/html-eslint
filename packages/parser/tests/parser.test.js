"use strict";

const createParser = require("../lib/parser");

describe("parser", () => {
  let parser = null;

  beforeEach(() => {
    parser = createParser();
  });

  it("parse", () => {
    const code = "<html></html>";
    expect(() => parser.parse(code)).not.toThrow();
  });

  it("base", () => {
    const code = "<html></html>";
    expect(parser.parse(code)).toMatchSnapshot();
  });
});
