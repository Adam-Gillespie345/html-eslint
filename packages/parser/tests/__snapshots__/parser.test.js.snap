// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser DOCTYPE 1`] = `
Object {
  "attrs": Array [],
  "body": Array [],
  "childNodes": Array [
    Object {
      "attrs": Array [],
      "childNodes": Array [],
      "end": 18,
      "endTag": null,
      "loc": Object {
        "end": Object {
          "column": 18,
          "line": 2,
        },
        "start": Object {
          "column": 3,
          "line": 2,
        },
      },
      "name": "html",
      "publicId": "",
      "range": Array [
        3,
        18,
      ],
      "start": 3,
      "startTag": null,
      "systemId": "",
      "type": "documentType",
    },
    Object {
      "attrs": Array [],
      "childNodes": Array [],
      "end": 37,
      "endTag": Object {
        "end": 37,
        "loc": Object {
          "end": Object {
            "column": 10,
            "line": 4,
          },
          "start": Object {
            "column": 3,
            "line": 4,
          },
        },
        "range": Array [
          30,
          37,
        ],
        "start": 30,
      },
      "loc": Object {
        "end": Object {
          "column": 10,
          "line": 4,
        },
        "start": Object {
          "column": 3,
          "line": 3,
        },
      },
      "namespaceURI": "http://www.w3.org/1999/xhtml",
      "range": Array [
        21,
        37,
      ],
      "start": 21,
      "startTag": Object {
        "end": 27,
        "loc": Object {
          "end": Object {
            "column": 9,
            "line": 3,
          },
          "start": Object {
            "column": 3,
            "line": 3,
          },
        },
        "range": Array [
          21,
          27,
        ],
        "start": 21,
      },
      "tagName": "html",
      "type": "Html",
    },
  ],
  "comments": Array [],
  "end": 37,
  "endTag": null,
  "loc": Object {
    "end": Object {
      "column": 10,
      "line": 4,
    },
    "start": Object {
      "column": 3,
      "line": 2,
    },
  },
  "mode": "no-quirks",
  "range": Array [
    3,
    37,
  ],
  "start": 3,
  "startTag": null,
  "tokens": Array [],
  "type": "Program",
}
`;

exports[`parser attr 1`] = `
Object {
  "attrs": Array [],
  "body": Array [],
  "childNodes": Array [
    Object {
      "attrs": Array [
        Object {
          "end": 16,
          "loc": Object {
            "end": Object {
              "column": 16,
              "line": 2,
            },
            "start": Object {
              "column": 6,
              "line": 2,
            },
          },
          "name": "attr",
          "range": Array [
            6,
            16,
          ],
          "start": 6,
          "value": "foo",
        },
      ],
      "childNodes": Array [
        Object {
          "attrs": Array [],
          "childNodes": Array [],
          "end": 18,
          "endTag": null,
          "loc": Object {
            "end": Object {
              "column": 1,
              "line": 3,
            },
            "start": Object {
              "column": 17,
              "line": 2,
            },
          },
          "range": Array [
            17,
            18,
          ],
          "start": 17,
          "startTag": null,
          "type": "text",
          "value": "
",
        },
      ],
      "end": 24,
      "endTag": Object {
        "end": 24,
        "loc": Object {
          "end": Object {
            "column": 7,
            "line": 3,
          },
          "start": Object {
            "column": 1,
            "line": 3,
          },
        },
        "range": Array [
          18,
          24,
        ],
        "start": 18,
      },
      "loc": Object {
        "end": Object {
          "column": 7,
          "line": 3,
        },
        "start": Object {
          "column": 1,
          "line": 2,
        },
      },
      "namespaceURI": "http://www.w3.org/1999/xhtml",
      "range": Array [
        1,
        24,
      ],
      "start": 1,
      "startTag": Object {
        "end": 17,
        "loc": Object {
          "end": Object {
            "column": 17,
            "line": 2,
          },
          "start": Object {
            "column": 1,
            "line": 2,
          },
        },
        "range": Array [
          1,
          17,
        ],
        "start": 1,
      },
      "tagName": "div",
      "type": "Div",
    },
    Object {
      "attrs": Array [],
      "childNodes": Array [],
      "end": 25,
      "endTag": null,
      "loc": Object {
        "end": Object {
          "column": 1,
          "line": 4,
        },
        "start": Object {
          "column": 7,
          "line": 3,
        },
      },
      "range": Array [
        24,
        25,
      ],
      "start": 24,
      "startTag": null,
      "type": "text",
      "value": "
",
    },
  ],
  "comments": Array [],
  "end": undefined,
  "endTag": null,
  "loc": Object {
    "end": Object {
      "column": undefined,
      "line": undefined,
    },
    "start": Object {
      "column": undefined,
      "line": undefined,
    },
  },
  "mode": "quirks",
  "range": Array [
    undefined,
    undefined,
  ],
  "start": undefined,
  "startTag": null,
  "tokens": Array [],
  "type": "Program",
}
`;

exports[`parser base 1`] = `
Object {
  "attrs": Array [],
  "body": Array [],
  "childNodes": Array [
    Object {
      "attrs": Array [],
      "childNodes": Array [],
      "end": 19,
      "endTag": Object {
        "end": 19,
        "loc": Object {
          "end": Object {
            "column": 10,
            "line": 3,
          },
          "start": Object {
            "column": 3,
            "line": 3,
          },
        },
        "range": Array [
          12,
          19,
        ],
        "start": 12,
      },
      "loc": Object {
        "end": Object {
          "column": 10,
          "line": 3,
        },
        "start": Object {
          "column": 3,
          "line": 2,
        },
      },
      "namespaceURI": "http://www.w3.org/1999/xhtml",
      "range": Array [
        3,
        19,
      ],
      "start": 3,
      "startTag": Object {
        "end": 9,
        "loc": Object {
          "end": Object {
            "column": 9,
            "line": 2,
          },
          "start": Object {
            "column": 3,
            "line": 2,
          },
        },
        "range": Array [
          3,
          9,
        ],
        "start": 3,
      },
      "tagName": "html",
      "type": "Html",
    },
  ],
  "comments": Array [],
  "end": 19,
  "endTag": null,
  "loc": Object {
    "end": Object {
      "column": 10,
      "line": 3,
    },
    "start": Object {
      "column": 3,
      "line": 2,
    },
  },
  "mode": "quirks",
  "range": Array [
    3,
    19,
  ],
  "start": 3,
  "startTag": null,
  "tokens": Array [],
  "type": "Program",
}
`;
